### PR TITLE
Disable pprof endpoints by default

### PR DESCRIPTION
**Pull request by justicz**
The `/debug/pprof` endpoints should not be enabled by default since they can leak potentially sensitive information. This PR adds a flag in `config.json` to enable these endpoints.

### CODE REVIEW

**Comment by algobolson** [comment]
This seems reasonable to me as "should have always been off" and it's the correct default. Probably only internal developers actually use pprof, and it's okay if we need to turn it back on.

**Comment by justicz** [comment]
Yeah, I'm not sure what a configuration version bump would achieve in this case since we want the default value to be false.

File: daemon/algod/api/server/router.go [hunk 1]
```diff
@@ -117,12 +118,21 @@ func NewRouter(logger logging.Logger, node *node.AlgorandFullNode, shutdown <-ch
...
+		// Registers /debug/pprof handler under root path and under /urlAuth path
+		// to support header or url-provided token.
+		router.PathPrefix(pprofEndpointPrefix).Handler(http.DefaultServeMux)
+
+		urlAuthRouter := router.PathPrefix(urlAuthEndpointPrefix)
+		urlAuthRouter.PathPrefix(pprofEndpointPrefix).Handler(http.DefaultServeMux).Name(debugRouteName)
+	} else {
+		// pprof disabled, route to black hole
+		disabledHandler := func(w http.ResponseWriter, r *http.Request) {
+			http.Error(w, "Profiler disabled", http.StatusForbidden)
```
Comment (tsachiherman) [inline daemon/algod/api/server/router.go]: I think that the desired behavior here is to return the same output as if "there is nothing there" while logging the incident.

Comment (justicz) [reply daemon/algod/api/server/router.go]: Sure, we can just not route pprof requests at all, done.

**Comment by justicz** [comment]
Thanks, made that change, let's see if that was the only place it needs to be enabled...

**Pull request merged**